dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1595,6 +1595,11 @@ public Identifier getRelatedIdentifier(Contentlet contentlet,String relationship
 
             SearchResponse response;
             final boolean DONT_PULL_PARENTS = Boolean.FALSE;
+
+            if (limit <= 0) {
```

Comment:
rather than setting a parameter mutable I would rename the limit param by limitParam and use limit such as
```
final int limit = limitParam <= 0? MAX_LIMIT: limitParam;
```

Reply:
requested changes made